atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/metrics/LastSweptTsUpdateScheduler.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.metrics;
+
+import com.palantir.common.concurrent.NamedThreadFactory;
+import com.palantir.common.concurrent.PTExecutors;
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class LastSweptTsUpdateScheduler implements Closeable {
+    static final long DELAY = Duration.ofSeconds(30).toMillis();
+    private final ScheduledExecutorService executorService;
+    private final Callable<Void> task;
+
+    LastSweptTsUpdateScheduler(ScheduledExecutorService executorService, Callable<Void> task) {
```

Comment:
```suggestion
    LastSweptTsUpdateScheduler(ScheduledExecutorService executorService, Runnable task) {
```
This will also make some other stuff nicer as you won't need to have `Void` methods